vcd/resource_vcd_vapp_vm.go

```diff
@@ -238,6 +210,82 @@ func resourceVcdVAppVmCreate(d *schema.ResourceData, meta interface{}) error {
 	return resourceVcdVAppVmUpdate(d, meta)
 }
 
+// Adds existing org VDC network to VM network configuration
+// Returns configured OrgVDCNetwork for Vm, networkName, error if any occur
+func addVdcNetwork(d *schema.ResourceData, vdc govcd.Vdc, vapp govcd.VApp, vcdClient *VCDClient) (vdcNetworks []*types.OrgVDCNetwork, networkName string, err error) {
```

Comment:
I don't agree with having this function to be added to the code base. This will make it significantly harder to apply any dynamic configuration to a nic. Like e.g. IpAllocationMode which could be passed to function AddVM() in underlying go-vcloud-director library.

Reply:
I agree regarding this and refactor will happen outside this PR. We will review vapp,vm and network functionality which is quite old code and functionality wise.